diag_manager/fms_diag_object.F90

```diff
@@ -673,4 +686,141 @@ subroutine dump_diag_obj( filename )
   call mpp_error( FATAL, "You can not use the modern diag manager without compiling with -Duse_yaml")
 #endif
 end subroutine
+
+!> @brief Writes brief diagnostic field info to the log file.
+!! @details If the <TT>do_diag_field_log</TT> namelist parameter is .TRUE.,
+!!     then a line briefly describing diagnostic field is added to
+!!     the log file.  Normally users should not call this subroutine
+!!     directly, since it is called by register_static_field and
+!!     register_diag_field if do_not_log is not set to .TRUE..  It is
+!!     used, however, in LM3 to avoid excessive logs due to the
+!!     number of fields registered for each of the tile types.  LM3
+!!     code uses a do_not_log parameter in the registration calls,
+!!     and subsequently calls this subroutine to log field information
+!!     under a generic name.
+SUBROUTINE fms_log_field_info(module_name, field_name, axes, long_name, units,&
```

Comment:
This routine is only used by the new. Its pretty similar to the old but some info needed to come from the diag object, also it has the changes for the separator and writes the header instead of doing that in the init.

My thinking was that if theres a new register routine in a separate file it would make sense to add a new log routine too. I think I'll change it though, it will simplifiy the do_not_log logic as well if I just make some changes to the original and use that for both.